pkg/notaryv2/notaryv2.go

```diff
@@ -128,5 +136,118 @@ func (v *notaryV2Verifier) verifyOutcomes(outcomes []*notation.VerificationOutco
 }
 
 func (v *notaryV2Verifier) FetchAttestations(ctx context.Context, opts images.Options) (*images.Response, error) {
-	return nil, errors.Errorf("not implemented")
+	v.log.V(2).Info("fetching attestations", "reference", opts.ImageRef)
+
+	rcRepoReference, err := ref.New(opts.ImageRef)
+	if err != nil {
+		panic(err)
+	}
+
+	rcConfigHost := config.Host{
+		Name:     rcRepoReference.Registry,
+		Hostname: rcRepoReference.Registry,
+	}
+
+	rcClient := regclient.New(regclient.WithConfigHost(rcConfigHost))
+
+	rcRepoReferrers, err := rcClient.ReferrerList(ctx, rcRepoReference)
+	if err != nil {
+		msg := err.Error()
+		logger.Info("failed to fetch attestations", "error", msg)
+		if strings.Contains(msg, "MANIFEST_UNKNOWN: manifest unknown") {
+			return nil, fmt.Errorf("not found")
+		}
+
+		return nil, err
+	}
+	rcReferrersDescs := rcRepoReferrers.Descriptors
+	statements := make([]map[string]interface{}, 0)
+
+	certsPEM := combineCerts(opts)
```

Comment:
Its possible for `attestors` to be null, right? In this case, we should skip the signatures check.